### PR TITLE
feat: add unit tests for CLI canvases create command

### PR DESCRIPTION
**Pull request by Vukotije**
## Summary

  Adds unit test coverage for the `canvases create` CLI command and the HTTP redirect safety policy introduced in #3638.

  The tests follow the same pattern used by existing CLI command tests (`widgets_test.go`, `usage/get_test.go`) — mock the API with `httptest.NewServer`, call `Execute()` directly, and validate output and errors through a captured `bytes.Buffer`.

  ### What's covered

  **`pkg/cli/client_test.go`** — Tests the redirect policy that prevents the silent-failure bug. Verifies that same-method redirects (GET→GET, POST→POST via 307) pass
  through, while method-changing redirects (POST→GET via 301) are rejected with an actionable error.

  **`pkg/cli/commands/canvases/create_test.go`** — Tests both the name-based and file-based create paths, including:
  - Success output in all three formats (text, JSON, YAML)
  - Response validation catching empty/malformed API responses
  - Server errors propagating correctly
  - The original bug scenario: a 301-redirecting server silently converting POST→GET now fails loudly instead of exiting 0

  ## Test plan
  - [x] `make test PKG_TEST_PACKAGES=./pkg/cli/...` — all 55 tests pass

### CODE REVIEW

File: pkg/cli/commands/canvases/create_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+	}))
+	t.Cleanup(server.Close)
+
+	ctx, _ := newCreateCommandContextForTest(t, server, "text")
+	ctx.Args = []string{"my-canvas"}
+	cmd := &createCommand{}
+
+	err := cmd.Execute(ctx)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "server returned an empty response")
```
[the remark this answers is not in the document]
Comment (Vukotije) [reply pkg/cli/commands/canvases/create_test.go:75]: this is taken care of in another pr

File: pkg/cli/commands/canvases/create_test.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+	config.HTTPClient = &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			if len(via) > 0 && req.Method != via[0].Method {
+				return fmt.Errorf(
+					"refusing to follow redirect that changes method from %s to %s",
+					via[0].Method, req.Method,
+				)
+			}
+			return nil
+		},
```
[the remark this answers is not in the document]
Comment (Vukotije) [reply pkg/cli/commands/canvases/create_test.go:218]: the inline policy is intentional to avoid an import cycle

File: pkg/cli/commands/canvases/create_test.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+	"github.com/superplanehq/superplane/pkg/cli/core"
+	"github.com/superplanehq/superplane/pkg/openapi_client"
+)
+
+func TestCreateCommandPrintsCanvasOnSuccess(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, http.MethodPost, r.Method)
+		require.Equal(t, "/api/v1/canvases", r.URL.Path)
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write([]byte(`{"canvas":{"metadata":{"id":"abc-123","name":"my-canvas"}}}`))
```
Comment (forestileao) [inline pkg/cli/commands/canvases/create_test.go:24]: We are repeating
```go
        w.Header().Set("Content-Type", "application/json")
		_, _ = w.Write([]byte(`{"canvas":{"metadata":{"id":"abc-123","name":"my-canvas"}}}`))
```
in a lot of places. Maybe it would be good to add one test Setup for tests that has the same response.

Like we do in other parts of the code like `authentication_test.go`, where we have only one setup and many internal tests. The use case is different, but the structure would be similar

Comment (Vukotije) [reply pkg/cli/commands/canvases/create_test.go:24]: I extracted newCanvasCreateServer for the some tests since they all need the exact same mock server. The edge case tests still set up their own servers inline because each one returns something different. I think extracting those would add indirection without  actualy reducing duplication, we would end up with a bunch of one-off helpers that each get called once.
